molr-mole-core/src/main/java/io/molr/mole/core/support/OngoingMissionRun.java

```diff
@@ -41,4 +44,107 @@ public OngoingMissionRun and() {
                 .map(MissionState::result);
     }
 
+    public Result whenFinished(Duration timeout) {
```

Comment:
I am not sure if those 2 whenFinished() methods should be here... in which clause would they be used? I think they might only be needed in your ReturnHelper...

Reply:
You are right! I wrote them initially when I did not have the complete design on my mind. And later missed to remove them.